### ISSUE TITLE
Public/private

### ISSUE THREAD

**Issue by yarelismartin** [yarelismartin/INDIVIDUAL-ASSESSMENT-vocab-YOU-lary#36]
Users can mark an entry as public or private
 Create a new navigation item called "Community" and on click of this item, users can see all the public entries
 If an entry is public, anyone can READ it. If it is private, only the user who created it can see it.
 If an entry is public, only the user who created it can CRUD on it
 If an entry is public, any user can copy the entry to their own set of entries and then CRUD on the new/copied entry

Acceptance Criteria:

- [x]  need a community page in nav 
- [x]   add a public/private option in the vocab form 
- [x]   the username/profile pic? should only be displayed when in community view 
- [x]   if a card is yours on the community tab then you should see the edit/del buttons
 - [x]  if the card isn't yours then you should see an option to add the card to your entries 

Consider:
- [ ]  filter btn in view all vocab might not work the same since its calling getVocab
- [ ] search bar needs to be adjusted to only search for public ?
 

Dependencies:
- get all public cards 
- get user info 
- POST card into your entity

**Comment by yarelismartin** [comment]
Encountered an issue with show Vocabs. To get all public vocabs to show I made this event listener for community                         
```
if (e.target.id.includes('community')) {
      clearDom();
      console.warn('community clicked');
      getAllPublicVocab(uid).then((vocab) => showVocabs(vocab, uid));
    }
```

In showVocabs I had a getLanguage(uid) api call in order for the lang name to show up on the vocab cards. Since this api call was specific to the user I got an error when in the community tab because it wasn't able to fetch the language names that were public and not created by the user. I ended up having to make a new api call getAllLanguage() and this had no uid attached to it. 
```
const getAllLanguage = () => new Promise((resolve, reject) => {
  fetch(`${endpoint}/language.json`, {
    method: 'GET',
    headers: {
      'Content-Type': 'application/json',
    },
  })
    .then((response) => response.json())
    .then((data) => {
      if (data) {
        resolve(Object.values(data));
      } else {
        resolve([]);
      }
    })
    .catch(reject);
});
```

Not sure if this is the best approach because if I was dealing with larger data and multiple users it could negatively affect the speed of the app. Not sure what would be another approach I should take. 

-------------------------------------------------

Currently I don't have a seprate function for showPublicVocab I'm just re-using the showVocabs but if I did have a seperate one then instead of making an api call for getAllLanguages I could make a call for getAllPublicLanguages. This way we aren't fetching more data than we actually need. Or instead of another page I could create a conditional; based on the page I'm on.
